url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/model/CourseCollectionDBTXT.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package coursescheduleapp.model;
+
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ *
+ * @author Drew
+ */
+public class CourseCollectionDBTXT implements CourseCollectionDB{
+
+    
+
+	@Override
+	public List<String> getAllSemesters() {
+		// TODO Auto-generated method stub
+		List<String> semesters = new ArrayList<String>();
+		semesters.add("Spring 2010");
+		semesters.add("Fall 2010");
+		return semesters;
+	}
+	
+	@Override
+	public void setSemester(String semesterName) {
+		// TODO Auto-generated method stub
+		
+	}
+
+    
+    public List<String> getInstructors(String courseName) {
+
+    	List<String> ins = new ArrayList<String>();
+    	
+    	if (courseName.equals("AI"))
+    	{
+    		ins.add("Prof. Norvig");
+    	}
+    	else if (courseName.equals("Calc III"))
+    	{
+    		ins.add("Dr. Butler");
+    	}
+    	else if (courseName.equals("Machine Learning"))
+    	{
+    		ins.add("Drew");
+    		ins.add("Andrew Ng");
+    	}
+    	else
+    	{
+    		ins.add("BAD SHOULD NOT HAPPEN");
+    	}
+    	return ins;
+    }
+
+	@Override
+	public List<String> getAllCourseNames() {
+		List<String> courses = new ArrayList<String>();
+        courses.add("Choose Course");
+        courses.add("AI");
+        courses.add("Calc III");
+        courses.add("Machine Learning");
+        return courses;
+	}
+
+	
+
+	@Override
+	public List<Course> getEquivCourses(String courseName) {
+		List<Course> cse = new ArrayList<Course>();
+    	List<String> ins = new ArrayList<String>();
+    	if (courseName == "AI")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.